mlir/lib/Dialect/LLVMIR/IR/LLVMTypes.cpp

```diff
@@ -280,7 +280,7 @@ getPointerDataLayoutEntry(DataLayoutEntryListRef params, LLVMPointerType type,
   for (DataLayoutEntryInterface entry : params) {
     if (!entry.isTypeEntry())
       continue;
-    if (cast<LLVMPointerType>(entry.getKey().get<Type>()).getAddressSpace() ==
+    if (cast<LLVMPointerType>(cast<Type>(entry.getKey())).getAddressSpace() ==
```

Comment:
Pre-existing issue, but the mix of llvm::cast and unqualified cast in this file is weird...

Reply:
It's already marked as deprecated, so maybe it's time to drop it...